### PR TITLE
Switch `dedent` to `outdent`

### LINKED ISSUE #8143: Dependency `dedent` has bug
`dedent` is in `package.json`, I just found it's buggy, I'll check how it effect our code.

```console
$ node -p "`\u0070\u0072\u0065\u0074\u0074\u0069\u0065\u0072`"
prettier

$ node -p "require('dedent')`\u0070\u0072\u0065\u0074\u0074\u0069\u0065\u0072`"
\u0070\u0072\u0065\u0074\u0074\u0069\u0065\u0072

$ node -p "require('outdent').outdent`\u0070\u0072\u0065\u0074\u0074\u0069\u0065\u0072`"
prettier
```

### PR DESCRIPTION
**Pull request by fisker**
`dedent` is working fine in other places, but can't use in `tests/`, it can't work with escaped characters, more details see #8143.

But having two similar package may cause confusion.

Switching to `outdent`

Fixes #8143

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: scripts/build/bundler.js [hunk 1]
```diff
@@ -37,6 +39,33 @@ const EXTERNALS = [
   "graceful-fs",
 ];
 
+const entries = [
```
Comment (thorn0) [inline scripts/build/bundler.js:42]: What is this change doing in this PR?

Comment (fisker) [reply scripts/build/bundler.js:42]: The entries of `outdent` need point to commonjs build,  and I tried fix in 4 bundles, still not enough, seems it's
 imported everywhere, so I move them to commom alias.

File: scripts/build/bundler.js [hunk 2]
```diff
@@ -118,6 +147,9 @@ function getRollupConfig(bundle) {
 
   const babelConfig = getBabelConfig(bundle);
 
+  const alias = { ...bundle.alias };
+  alias.entries = [...entries, ...(alias.entries || [])];
```
Comment (lydell) [inline scripts/build/bundler.js:151]: Does this work?

```suggestion
  const alias = { ...bundle.alias, entries: [...entries, ...(bundle.entries || [])] };
```

Comment (fisker) [reply scripts/build/bundler.js:151]: Yes, I think so.

Comment (fisker) [reply scripts/build/bundler.js]: Wait, no... bundle.alias can be undefined.

**Comment by thorn0** [comment]
Would be cool to replace this with a build-time transformation (Babel plugin) one day. :)

**Pull request merged**